x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/job/persistence/JobResultsProviderTests.java

```diff
@@ -897,8 +897,9 @@ public void testDatafeedTimingStats_EmptyJobList() {
         JobResultsProvider provider = createProvider(client);
         provider.datafeedTimingStats(
             List.of(),
-            statsByJobId -> assertThat(statsByJobId, anEmptyMap()),
-            e -> { throw new AssertionError(); });
+            ActionListener.wrap(
+                statsByJobId -> assertThat(statsByJobId, anEmptyMap()),
+                e -> { throw new AssertionError(); }));
```

Comment:
I know you didn't introduce this problem, but if this assertion error is ever thrown it will be frustrating because it will not say what the root case was.  So please can you change it to `e -> { throw new AssertionError("Failure getting datafeed timing stats", e); }`.

Reply:
👍